autotest.rb

```diff
@@ -17,18 +17,27 @@
     
     $stdout.puts "Changes:\n  " + rerun.join("\n  ")
     $stdout.write "Forking spec..."
-    
-    Process.waitpid(fork do
+
+    if (/cygwin|mswin|mingw|bccwin|wince|emx/ =~ RUBY_PLATFORM) != nil
```

Comment:
There are a number of gems that offer this "os" functionality but rather than pull in another gem I went for this solution from Aaron Hinni : http://stackoverflow.com/questions/170956/how-can-i-find-which-operating-system-my-ruby-program-is-running-on